### PR TITLE
Deprecate composable-map

### PR DESCRIPTION
**Pull request by puneet-pdx**
Changes - 
- remove dependency of composable-map from authentication-app, as it is not used there
- remove dependency of composable-map from compass-app. Add dependency on geo-compose and update implementation to use composable mapview fun
- remove dependency of composable-map from floor-filter-app. Add dependency on geo-compose and update implementation to use composable mapview fun
- add deprecation notice to composable-map ReadMe file, logger warning to build.gradle.kts and deprecated annotation to the ComposableMap fun 

- remove dependency of composable-map from template-app. Add dependency on geo-compose and update implementation to use composable mapview fun

### CODE REVIEW

**Review by hud10837 — CHANGES_REQUESTED** [review]
Apps work as expected, I just think we should add deprecation warnings to everything

File: toolkit/composable-map/src/main/java/com/arcgismaps/toolkit/composablemap/ComposableMap.kt [hunk 1]
```diff
@@ -39,6 +39,7 @@ import com.arcgismaps.mapping.ViewpointType
 import com.arcgismaps.mapping.view.MapView
 import kotlinx.coroutines.launch
 
+@Deprecated("Use the new composable MapView fun available in the geo-compose toolkit component instead")
```
[the remark this answers is not in the document]
Comment (puneet-pdx) [reply toolkit/composable-map/src/main/java/com/arcgismaps/toolkit/composablemap/ComposableMap.kt:42]: I don't know if we want to add the annotation to all individual components that are used with the `ComposableMap` function. We have added the deprecation notice to the module readme and to the main component in the module. I think that should be enough.
@rchintapalli @gunt0001

[the remark this answers is not in the document]
Comment (puneet-pdx) [reply toolkit/composable-map/src/main/java/com/arcgismaps/toolkit/composablemap/ComposableMap.kt:42]: I have updated it to say `function`

**Review by hud10837 — APPROVED** [review]
Approving this so we can get it in before holistic, but I think we should still consider the other deprecations.

**Comment by puneet-pdx** [comment]
> Approving this so we can get it in before holistic, but I think we should still consider the other deprecations.

@hud10837 I have added other deprecations

File: microapps/FloorFilterApp/app/src/main/java/com/arcgismaps/toolkit/floorfilterapp/screens/MainScreen.kt [hunk 2]
```diff
@@ -21,33 +21,93 @@ import androidx.compose.foundation.layout.Box
...
+                    val floorFilterSelectionType =
+                        floorFilterSelection.type as FloorFilterSelection.Type.FloorFacility
+                    floorFilterSelectionType.facility.geometry?.let {
+                        mapViewpointOperation =
+                            MapViewpointOperation.Set(Viewpoint(getEnvelopeWithBuffer(it)))
+                    }
+                }
 
-    val mapViewModel = viewModel<MapViewModel>(factory = MapViewModelFactory(floorAwareWebMap))
+                else -> {}
```
Comment (eri9000) [inline microapps/FloorFilterApp/app/src/main/java/com/arcgismaps/toolkit/floorfilterapp/screens/MainScreen.kt:86]: Log something to the screen instead?

**Review by eri9000 — APPROVED** [review]
a few things, otherwise LGTM

File: microapps/FloorFilterApp/app/src/main/java/com/arcgismaps/toolkit/floorfilterapp/screens/MainScreen.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (puneet-pdx) [reply microapps/FloorFilterApp/app/src/main/java/com/arcgismaps/toolkit/floorfilterapp/screens/MainScreen.kt:86]: That won't be of any help, as the third FloorFilterSelection.Type is FloorLevel and we are not setting the viewpoint related to the level for this example. This block is just to show the user how they can refocus the viewpoint of the map to the selected site and facility. We left the level out of it.

**Pull request merged**